src/classes/HiFiAudioAPIData.ts

```diff
@@ -32,6 +32,188 @@ export class Point3D {
+
+/**
+ * HiFi Spatial Audio uses a right-handed Cartesian coordinate system
+ * with FORWARD pointing along negative Z-axis and UP along positive Y-axis.
+ *
+ * If the World uses a different coordinate system then it will be necessary
+ * to convert World-frame Position into the HiFi-frame.
+ * HiFiCoordinateFrameUtil is a helper class for this scenario.
+ *
+ * To build a HiFiCoordinateFrameUtil instance you must present three arguments:
+ * (1) The normalized FORWARD axis in the World-frame
+ * (2) The normalized UP axis in the World-frame
+ * (3) Whether it is a RIGHT- or LEFT-handed coordinate system
+ *
+ * Note: there is no need to transform Orientation into the HiFi-frame:
+ * its value remains unchanged between the two systems.
+ *
+ */
+export class HiFiCoordinateFrameUtil {
+    _worldToHifi: Matrix3;
+    _hifiToWorld: Matrix3;
+
+    /**
+     * Construct a HiFiCoordinateFrameUtil to easily transform into and out of the coordinate frame
+     * used for HiFi Spatial Audio calculations.
+     *
+     * @param forward - the World-frame FORWARD direction.
+     * @param up - the World-frame Up direction.
+     * @param isLeft - true if World-frame is a left-handed coordinate frame, else false (the default)
+     *
+     * How to know if your coordinate frame is left-handed?
+     * If Z = vector_cross_product(X, Y) as per the right-hand rule
+     * (from your physics or 3D geometry class)
+     * then it is right-handed, else it must be left-handed.
+     */
+    constructor(forward: Point3D, up: Point3D, isLeft = false) {
+        // We assemble a Matrix3: worldToCanonical which transforms World-frame Point3Ds
+        // to a hypothentical canonical-frame (e.g. fwd=x-axis, up=y-axis, right=z-axis).
+        // It is just the matrix of the three cardinal directions as rows:
+        //
+        // | <-forward-> |
+        // | <-  up   -> |
+        // | <- right -> |
+        //
+        let right = Point3D.cross(forward, up);
+        if (isLeft) {
+            // Note: for left-handed systems we reflect 'right' to actually be 'left'
+            right.negate();
+        }
+        let worldToCanonical = new Matrix3(forward, up, right);
+
+        // Similarly we create another Matrix3: canonicalToHifi
+        // using HiFi's directions (forward, up, and right)
+        // but this time we transpose it to get the inverse.
+        let canonicalToHifi = new Matrix3(
+            new Point3D({x:0, y:0, z:-1}),
+            new Point3D({x:0, y:1, z:0}),
+            new Point3D({x:1, y:0, z:0}));
+        canonicalToHifi.transpose();
+
+        // The final Matrix is: canonicalToHifi * worldToCanonical 
+        this._worldToHifi = Matrix3.multiply(canonicalToHifi, worldToCanonical);
+
+        // For convenience we cache the transposed matrix for transforming in the other direction
+        this._hifiToWorld = new Matrix3(this._worldToHifi.a, this._worldToHifi.b, this._worldToHifi.c);
+        this._hifiToWorld.transpose();
+    }
```

Comment:
This is untested in TS, however I first implemented the algorithm in matlab for easy calculations and inspection and verified it worked for going from Unity left-handed coordinate frame to HiFi's right-handed system.